src/schism/schism_esmf_util.F90

```diff
@@ -1366,74 +1385,123 @@ subroutine SCHISM_MeshCreateElement(comp, kwe, rc)
   ! As the list of owned and non-owned nodes is not preserved in the ESMF_Mesh
   ! structure, we need to save this information to an internal state, for later 
   ! use in Array/Field creation.
-  !call ESMF_GridCompGetInternalState(comp, internalState, localrc)
-  !_SCHISM_LOG_AND_FINALIZE_ON_ERROR_(rc_)
+  call ESMF_GridCompGetInternalState(comp, internalState, localrc)
+  _SCHISM_LOG_AND_FINALIZE_ON_ERROR_(rc_)
 
-  !isDataPtr => internalState%wrap
-
-  !isDataPtr%numOwnedNodes = 0
-  !isDataPtr%numForeignNodes = 0
-  !do ip = 1, npa
-  !  ! non-ghost nodes
-  !  if (ip <= np) then
-  !    if (nodeowners(ip) == localPet) then 
-  !      isDataPtr%numOwnedNodes = isDataPtr%numOwnedNodes + 1
-  !    else 
-  !      isDataPtr%numForeignNodes = isDataPtr%numForeignNodes + 1
-  !    end if
-  !  end if
-  !end do
+  isDataPtr => internalState%wrap
 
-  !allocate(isDataPtr%ownedNodeIds(isDataPtr%numOwnedNodes), stat=localrc)
-  !allocate(isDataPtr%foreignNodeIds(isDataPtr%numForeignNodes), stat=localrc)
-
-  !ownedCount = 0
-  !foreignCount = 0
-  !do ip = 1, npa
-  !  ! non-ghost nodes
-  !  if (ip <= np) then
-  !    if (nodeowners(ip) == localPet) then
-  !      ownedCount=ownedCount + 1
-  !      isDataPtr%ownedNodeIds(ownedCount) = ip
-  !    else
-  !      foreignCount=foreignCount + 1
-  !      isDataPtr%foreignNodeIds(foreignCount) = ip
-  !    endif
-  !  end if
-  !enddo
+  isDataPtr%numOwnedNodes = 0
+  isDataPtr%numForeignNodes = 0
+  do ip = 1, npa
+    ! non-ghost nodes
+    if (ip <= np) then
+      if (nodeowners(ip) == localPet) then 
+        isDataPtr%numOwnedNodes = isDataPtr%numOwnedNodes + 1
+      else 
+        isDataPtr%numForeignNodes = isDataPtr%numForeignNodes + 1
+      end if
+    end if
+  end do
+
+  allocate(isDataPtr%ownedNodeIds(isDataPtr%numOwnedNodes), stat=localrc)
+  allocate(isDataPtr%foreignNodeIds(isDataPtr%numForeignNodes), stat=localrc)
+
+  ownedCount = 0
+  foreignCount = 0
+  do ip = 1, npa
+    ! non-ghost nodes
+    if (ip <= np) then
+      if (nodeowners(ip) == localPet) then
+        ownedCount=ownedCount + 1
+        isDataPtr%ownedNodeIds(ownedCount) = ip
+      else
+        foreignCount=foreignCount + 1
+        isDataPtr%foreignNodeIds(foreignCount) = ip
+      endif
+    end if
+  enddo
 
   ! add metadata
-  !field = ESMF_FieldEmptyCreate(name='mesh_topology', rc=localrc)
-  !_SCHISM_LOG_AND_FINALIZE_ON_ERROR_(rc_) 
+  field = ESMF_FieldEmptyCreate(name='mesh_topology', rc=localrc)
+  _SCHISM_LOG_AND_FINALIZE_ON_ERROR_(rc_) 
 
-  !call ESMF_AttributeSet(field, 'cf_role', 'mesh_topology', rc=localrc)
-  !_SCHISM_LOG_AND_FINALIZE_ON_ERROR_(rc_)
+  call ESMF_AttributeSet(field, 'cf_role', 'mesh_topology', rc=localrc)
```

Comment:
We should use the ESMF_Info structure, i.e. InfoSet (but maybe not in this PR, but at a later stage)

Reply:
@platipodium Yes, it would be better to address it in an other PR.